src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -171,10 +171,13 @@ public void addTag(Tag t) throws UniqueTagList.DuplicateTagException {
      * @throws TagNotFoundException
      * @throws IllegalValueException
      */
-    public void removeTag(String tagGettingRemoved) throws TagNotFoundException, IllegalValueException {
+    public void removeTag(String tagGettingRemoved) throws TagNotFoundException,
```

Comment:
How would this method throw a `PersonNotFoundException`?

Reply:
This is because of the `persons.setPerson` method used. This exception will not be thrown but if i dont write it here the project wont build.

Reply:
If it is reasonable to assume that this method will only deal with persons that exist, I suggest catching the `PersonNotFoundException` specifically and cause an assertion to indicate that it is an "impossible" condition.